templates/web/src/sdk.ts.twig

```diff
@@ -137,71 +143,103 @@ class {{ spec.title | caseUcfirst }} {
+            this.realtime.timeout = window?.setTimeout(() => {
+                this.realtime.createSocket();
+            }, 50);
+        },
         createSocket: () => {
+            if (this.realtime.channels.size < 1) return;
             const channels = new URLSearchParams();
             channels.set('project', this.config.project);
-            for (const property in this.realtime.channels) {
-                channels.append('channels[]', property);
-            }
-            if (this.realtime.socket?.readyState === WebSocket.OPEN) {
-                this.realtime.socket.close();
-            }
-
-            this.realtime.socket = new WebSocket(this.config.endpointRealtime + '/realtime?' + channels.toString());
-            this.realtime.socket?.addEventListener('message', this.realtime.authenticate);
+            this.realtime.channels.forEach(channel => {
+                channels.append('channels[]', channel);
+            });
 
-            for (const channel in this.realtime.channels) {
-                this.realtime.channels[channel].forEach(callback => {
-                    this.realtime.socket?.addEventListener('message', callback);
-                });
-            }
+            const url = this.config.endpointRealtime + '/realtime?' + channels.toString();
 
-            this.realtime.socket.addEventListener('close', event => {
-                if (this.realtime?.lastMessage?.type === 'error' && (<RealtimeResponseError>this.realtime?.lastMessage.data).code === 1008) {
-                    return;
+            if(url !== this.realtime.url || !this.realtime.socket || this.realtime.socket?.readyState > WebSocket.OPEN) {
+                if (this.realtime.socket && [WebSocket.OPEN, WebSocket.CONNECTING].includes(this.realtime.socket?.readyState)) {
+                    this.realtime.reconnect = false;
+                    this.realtime.socket.close();
                 }
-                console.error('Realtime got disconnected. Reconnect will be attempted in 1 second.', event.reason);
-                setTimeout(() => {
-                    this.realtime.createSocket();
-                }, 1000);
-            })
+
+                this.realtime.url = url;
+                this.realtime.socket = new WebSocket(url);
+                this.realtime.socket.addEventListener('message', this.realtime.onMessage);
+
+                this.realtime.socket.addEventListener('close', event => {
+                    if (!this.realtime.reconnect || (this.realtime?.lastMessage?.type === 'error' && (<RealtimeResponseError>this.realtime?.lastMessage.data).code === 1008)) {
+                        this.realtime.reconnect = true;
+                        return;
+                    }
+                    console.error('Realtime got disconnected. Reconnect will be attempted in 1 second.', event.reason);
+                    setTimeout(() => {
+                        this.realtime.createSocket();
+                    }, 1000);
```

Comment:
On the weekend Eldad had an idea to slow it down gradually, can we do that? For example:

1st - 5th request: Every 1 second
5th - 15th request: Evety 5 seconds
15th - 100th request: Every 10 seconds
100th+ - Every 60 seconds

Reply:
Oh yeah!

Forgot about that, going to add 🙏🏻

Reply:
Added 👍🏻